### PR TITLE
injectproxy: allow multiple enforcement matchers

### PR DESCRIPTION
**Pull request by s-urbaniak**
Currently, the SetRecursive method does not work with multiple matchers.
This fixes it by introducing a small intermediate struct
and adds unit tests.

Signed-off-by: Sergiusz Urbaniak <sergiusz.urbaniak@gmail.com>

### CODE REVIEW

**Comment by s-urbaniak** [comment]
I missed one edge case, looking into it.

**Comment by s-urbaniak** [comment]
@simonpasquier edge case fixed, PTAL :-)

**Comment by s-urbaniak** [comment]
cc'ing @lilic @brancz @paulfantom for a broader review, PTAL :-)

File: injectproxy/enforce.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	return &Enforcer{
+		labelMatchers: entries,
+	}
+}
+
+// EnforceNode walks the given node recursively
+// and enforces the given label enforcer on it.
+//
+// Whenever a parser.MatrixSelector or parser.VectorSelector AST node is found,
+// their label enforcer are being potentially modified.
```
Comment (squat) [inline injectproxy/enforce.go:43]: tiny grammar nit: we need correspondence between `are` and the singular `label enforcer`

File: injectproxy/enforce.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+//
+// If a node label matcher equals the name with one of the given enforcer,
```
Comment (squat) [inline injectproxy/enforce.go:45]: This comment is a bit hard for me to understand. Can we rephrase? Maybe:
```
If a node's label matcher has the same name as a label matcher of the given enforcer, then it will be replaced.
```

File: injectproxy/enforce.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		labelMatchers: entries,
+	}
+}
+
+// EnforceNode walks the given node recursively
+// and enforces the given label enforcer on it.
+//
+// Whenever a parser.MatrixSelector or parser.VectorSelector AST node is found,
+// their label enforcer are being potentially modified.
+//
```
Comment (squat) [inline injectproxy/enforce.go:44]: Let's remove this newline, since the next sentence actually refines the sentence above.

File: injectproxy/enforce.go (hunk 1 above)
Comment (squat) [reply injectproxy/enforce.go:43]: Is it the label enforcer of the node that is being modified? Or the label matcher of the node?

Comment (s-urbaniak) [reply injectproxy/enforce.go:43]: @squat ah, thank you for grammar corrections :pray: it is the latter, the label matcher of node. The enforcer is immutable it is "configuration data" if that makes sense.

File: injectproxy/enforce.go (hunk 2 above)
Comment (s-urbaniak) [reply injectproxy/enforce.go:45]: sounds good :+1:

File: injectproxy/enforce.go (hunk 3 above)
Comment (s-urbaniak) [reply injectproxy/enforce.go:44]: fixed

**Review by brancz — APPROVED** [review]
lgtm but giving others another chance to review

**Review by lilic — APPROVED** [review]
lgtm
